### ISSUE TITLE
Enums as Theory values cause tests to fail/not run in aspnetcore project

### ISSUE THREAD

**Issue by rp0m** [xunit/xunit#1903]
Hi all,

I've been trialling xUnit for our project over the last few days. So far so good, except for a strange enum + theory issue that completely derailed me. I've set up a tiny aspnetcore 2.1 project here: https://github.com/rp0m/xUnit_TheoryEnumIssue_AspDotNetCore

The issue is basically tests not running/hanging/hanging visual studio/test explorer when using an enum, declared in a separate project within a theory (basically this: https://github.com/rp0m/xUnit_TheoryEnumIssue_AspDotNetCore/blob/master/xUnit_Test.Test/NumbersTest.cs).

What's interesting is that two out of five tests do run okay - the Fact and first Theory, which casts enums to an int. 

Moreover, the issue does not seem to be present when declaring and using an enum inside a single assembly. For example - if I create a test and a sample enum in that test, then repeat this process, then everything runs okay.

I get all sorts of weird issues due to this - from problematic tests not executing (theory + enum) to visual studio/test explorer hanging.

The workaround is to cast the enum to an int in the [InlineData] constructor, however it is not an ideal solution for obvious reasons.

**Comment by bradwilson** [comment]
This appears to be a problem with your test project.

I made the following changes to `xUnit_Test.Test.csproj` (to match `xUnit_Test.csproj`):

- Convert Project SDK to `Microsoft.NET.Sdk.Web`
- Convert target framework to `netcoreapp2.1`
- Replace explicit ASP.NET Core references with `Microsoft.AspNetCore.App` 

After making these changes, the tests ran as expected:

![image](https://user-images.githubusercontent.com/16944/52548819-a7766d00-2d84-11e9-86a3-00b8b9375eeb.png)

**Comment by bradwilson** [comment]
If I had to guess, I'd say the failure to load the Enum values was because the dependencies weren't being properly satisfied when you referenced the ASP.NET Core project. It worked fine when the enums were defined in the tests, because then dependencies became irrelevant.

**Comment by rp0m** [comment]
Hi Brad,

Thanks for your help! Just to address your points:

* Convert Project SDK to `Microsoft.NET.Sdk.Web`. What effect does this have? The original value is specified in VS2017 xUnit project template, so I that's why I had it.
* Convert target framework to `netcoreapp2.1`. This was a mistake I made in the test app. Good point :)
* Replace explicit ASP.NET Core references with `Microsoft.AspNetCore.App`. I've had a bunch of issues around dependencies that I was able to work around by adding explicit references previously. Rolling back to `Microsoft.AspNetCore.App` seemed to work okay now. Not sure what my initial issues were.

Long story short - all tests now run as expected. Once again, thanks for your help!

**Comment by bradwilson** [comment]
The `Microsoft.NET.Sdk.Web` change is what makes the version-less `Microsoft.AspNetCore.App` work.

**Comment by sandves** [comment]
@bradwilson I am facing the exact same issue as @rp0m.

I also managed to get the tests working by adding a package reference to `Microsoft.AspnetCore.App`.

However, I do not think this is a good solution. Why should my unit test project have a dependency on all of the packages in `Microsoft.AspnetCore.App` when none of my tests has anything to do with Web? It does not make sense to me. See [demo repository](https://github.com/sandves/bug-demo).

I also made an identical test using NUnit and the `TestCase` attribute. This works as expected without adding a reference to `Microsoft.AspnetCore.App`.

**Comment by bradwilson** [comment]
@sandves You may have a different issue, so opening a new bug makes more sense than piling onto this already closed one. Please open a new issue.
